libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwner.kt

```diff
@@ -0,0 +1,176 @@
+package com.mapbox.navigation.core.lifecycle
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.utils.internal.LoggerProvider.logger
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class CarAppLifecycleOwner : LifecycleOwner {
+
+    // Keeps track of the activities created and foregrounded
+    private var activitiesCreated = 0
+    private var activitiesForegrounded = 0
+
+    // Keeps track of the car session created and foregrounded
+    private var lifecycleCreated = 0
+    private var lifecycleForegrounded = 0
+
+    // Keeps track of the activities changing configurations
+    private var createdChangingConfiguration = 0
+    private var foregroundedChangingConfiguration = 0
+
+    private val lifecycleRegistry = LifecycleRegistry(this)
+        .apply { currentState = Lifecycle.State.INITIALIZED }
+
+    @VisibleForTesting
+    internal val startedReferenceCounter = object : DefaultLifecycleObserver {
+        override fun onCreate(owner: LifecycleOwner) {
+            lifecycleCreated++
+            logger.i(TAG, Message("car onCreate"))
```

Comment:
```suggestion
            logger.i(TAG, Message("lifecycle owner ($owner) onCreate"))
```
let's make this log clearer, same everywhere below. Having access to string representation or hash code will help us debug.

Reply:
yeah, it used to only allow a "car" lifecycle to attach. BUt now it's any LifecycleOwner. So this was stale, nice catch!